### ISSUE TITLE
AssertionError in substituteMethodReturnType

### ISSUE THREAD

**Issue by panacekcz** [typetools/checker-framework#1411]
This error appeared when testing a modified index checker on plume-lib, in [MultiVersionControl.java](https://github.com/mernst/plume-lib-check-index-new/blob/3f82f73faaf4c01bee92944a1ea16d8e854bf2fb/java/src/plume/MultiVersionControl.java#L887). This report also uses index checker to trigger this error through an inferred `@SameLen` qualifier in the same way, but the error seems to be in the framework.
Minimalized test case:
```java
import org.checkerframework.dataflow.qual.Pure;
interface IGeneric<V>{
	@Pure public V get();
}
interface IConcrete extends IGeneric<char[]>{
}
public class SubstituteCrash {
	static void m(IConcrete ic) {
		char[] val = ic.get();
	}
}
```
Run the checker as:
```
checker/bin/javac -processor org.checkerframework.checker.index.IndexChecker -AprintErrorStack  SubstituteCrash.java
```
Error output:
```
error: SourceChecker.typeProcess: unexpected Throwable (AssertionError) while processing SubstituteCrash.java
  Compilation unit: SubstituteCrash.java
  Exception: java.lang.AssertionError; Stack trace: org.checkerframework.javacutil.InternalUtils.substituteMethodReturnType(InternalUtils.java:392)
  org.checkerframework.framework.util.FlowExpressionParseUtil.parseMethod(FlowExpressionParseUtil.java:541)
  org.checkerframework.framework.util.FlowExpressionParseUtil.parseHelper(FlowExpressionParseUtil.java:166)
  org.checkerframework.framework.util.FlowExpressionParseUtil.parseMemberSelect(FlowExpressionParseUtil.java:219)
  org.checkerframework.framework.util.FlowExpressionParseUtil.parseHelper(FlowExpressionParseUtil.java:168)
  org.checkerframework.framework.util.FlowExpressionParseUtil.parse(FlowExpressionParseUtil.java:131)
  org.checkerframework.framework.type.GenericAnnotatedTypeFactory.getReceiverFromJavaExpressionString(GenericAnnotatedTypeFactory.java:757)
  org.checkerframework.checker.index.samelen.SameLenTransfer.propagateCombinedSameLen(SameLenTransfer.java:134)
  org.checkerframework.checker.index.samelen.SameLenTransfer.visitAssignment(SameLenTransfer.java:110)
  org.checkerframework.checker.index.samelen.SameLenTransfer.visitAssignment(SameLenTransfer.java:35)
  org.checkerframework.dataflow.cfg.node.AssignmentNode.accept(AssignmentNode.java:63)
  org.checkerframework.dataflow.analysis.Analysis.callTransferFunction(Analysis.java:404)
  org.checkerframework.dataflow.analysis.Analysis.performAnalysis(Analysis.java:200)
  org.checkerframework.framework.type.GenericAnnotatedTypeFactory.analyze(GenericAnnotatedTypeFactory.java:1102)
  org.checkerframework.framework.type.GenericAnnotatedTypeFactory.analyze(GenericAnnotatedTypeFactory.java:1059)
  org.checkerframework.framework.type.GenericAnnotatedTypeFactory.performFlowAnalysis(GenericAnnotatedTypeFactory.java:989)
  org.checkerframework.framework.type.GenericAnnotatedTypeFactory.checkAndPerformFlowAnalysis(GenericAnnotatedTypeFactory.java:1320)
  org.checkerframework.framework.type.GenericAnnotatedTypeFactory.preProcessClassTree(GenericAnnotatedTypeFactory.java:224)
  org.checkerframework.common.basetype.BaseTypeVisitor.visitClass(BaseTypeVisitor.java:281)
  org.checkerframework.common.basetype.BaseTypeVisitor.visitClass(BaseTypeVisitor.java:160)
  com.sun.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:720)
  com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:50)
  org.checkerframework.framework.source.SourceVisitor.visit(SourceVisitor.java:66)
  org.checkerframework.framework.source.SourceChecker.typeProcess(SourceChecker.java:960)
  org.checkerframework.common.basetype.BaseTypeChecker.typeProcess(BaseTypeChecker.java:494)
  org.checkerframework.common.basetype.BaseTypeChecker.typeProcess(BaseTypeChecker.java:487)
  org.checkerframework.javacutil.AbstractTypeProcessor$AttributionTaskListener.finished(AbstractTypeProcessor.java:185)
  com.sun.tools.javac.api.ClientCodeWrapper$WrappedTaskListener.finished(ClientCodeWrapper.java:681)
  com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:111)
  com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1342)
  com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1296)
  com.sun.tools.javac.main.JavaCompiler.compile2(JavaCompiler.java:901)
  com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:860)
  com.sun.tools.javac.main.Main.compile(Main.java:523)
  com.sun.tools.javac.main.Main.compile(Main.java:381)
  com.sun.tools.javac.main.Main.compile(Main.java:370)
  com.sun.tools.javac.main.Main.compile(Main.java:361)
  com.sun.tools.javac.Main.compile(Main.java:56)
  com.sun.tools.javac.Main.main(Main.java:42)
1 error
```
In [`FlowExpressionParseUtil.parseMethod`](https://github.com/typetools/checker-framework/blob/2a69a2d32daaf6bd536c6214df3299c03ab1b6d4/framework/src/org/checkerframework/framework/util/FlowExpressionParseUtil.java#L541), `methodElement` is `IGeneric.get`, and its return type is a type variable. However the type of the receiver is `IConcrete`, which has no type parameters, so in [`substituteMethodReturnType`](https://github.com/typetools/checker-framework/blob/2a69a2d32daaf6bd536c6214df3299c03ab1b6d4/javacutil/src/org/checkerframework/javacutil/InternalUtils.java#L386), no type parameter is found to substitute.
